creatorkit/Toolbox.md

```diff
@@ -150,24 +150,36 @@ If your Clickers are powered on, they should be displayed on this page. You can
 
 ![ProvisionClicker3](../images/provision_3.png)
 
-This step is required to make sure that only 6LoWPAN devices you choose are able to access your datastores via Ci40.
 
-## Building Code from source
+## Building from source code
 
 Note: You do not need to do this to set up and run the example projects for Creator Kit. If you want to expand on the projects and eventually build your own it is recommended that you set up the first project and then come back to this section once you have seen the system in action. You can then move on to building your own IoT project using Creator Kit.
 
+### Installing dependencies
+
 Initially you need to install dependencies required to build. run the following the terminal on your Ubuntu build machine
 
-<code>$ sudo apt-get -y install g++ gcc binutils bzip2 flex python perl make libncurses5-dev libssl-dev grep unzip gawk subversion zlib1g-dev build-essential git mercurial cmake python-nose python-lxml</code>
+<code>$ sudo apt-get -y install g++ gcc binutils bzip2 flex python perl make libncurses5-dev libncursesw5-dev libssl-dev grep unzip gawk subversion zlib1g-dev build-essential git mercurial cmake python-nose python-lxml</code>
+
+### Installing Contiki compiler
+
+In order to build Contiki code for Clicker boards you will need the XC32 compiler from Microchip. For 64-bit Ubuntu, 32-bit runtime libraries must be installed before the XC32 compiler can be run:
+
+<code>$ sudo apt-get install libc6:i386</code>
+
+Follow the official Linux installation instructions [here](http://microchip.wikidot.com/xc32:installation) to install the latest version.
+
+### Installing repo tool
 
 If you do not have repo installed run the following
 
 <code>$ curl https://storage.googleapis.com/git-repo-downloads/repo > ~/bin/repo</code>
 
 <code>$ chmod a+x ~/bin/repo</code>
 
+## Getting and building the code
 
-You should now be able to run the following in a directory of your choice.
+The following commands will sync to the master branch of the source repositories. To sync to a different branch append " -b <branch-name>" to the "repo init" command.
 
 <code>$ repo init -u  https://github.com/CreatorKit/manifest.git</code>
```

Comment:
It is better to give the complete example command

repo init -u  https://github.com/CreatorKit/manifest.git -b <branch-name>